src/Infrastructure/Controller/Admin/UserCrudController.php

```diff
@@ -0,0 +1,109 @@
+{
+    public function __construct(
+        private readonly UserPasswordHasherInterface $userPasswordHasher,
+        private readonly IdFactoryInterface $idFactory,
+    ) {
+    }
+
+    public static function getEntityFqcn(): string
+    {
+        return User::class;
+    }
+
+    public function createEntity(string $entityFqcn): User
+    {
+        return new User($this->idFactory->make());
+    }
+
+    public function configureCrud(Crud $crud): Crud
+    {
+        return $crud
+            ->setEntityLabelInSingular('Utilisateur')
+            ->setEntityLabelInPlural('Utilisateurs')
+        ;
+    }
+
+    public function configureFields(string $pageName): iterable
+    {
+        $fields = [
+            TextField::new('fullName')->setLabel('Prénom / Nom'),
+            EmailField::new('email'),
+            AssociationField::new('organizations')
+                ->setFormTypeOption('by_reference', false)
+                ->setFormTypeOption('choice_label', 'name')
+                ->setLabel('Organisation(s)')
+                ->formatValue(function ($value, $entity) {
+                    return implode(', ', $entity->getOrganizations()->toArray());
+                }),
+        ];
+
+        $password = TextField::new('password')
+            ->setFormType(RepeatedType::class)
+            ->setFormTypeOptions([
+                'type' => PasswordType::class,
+                'first_options' => ['label' => 'Mot de passe'],
+                'second_options' => ['label' => 'Répéter le mot de passe'],
+                'mapped' => false,
+            ])
+            ->setRequired($pageName === Crud::PAGE_NEW)
+            ->onlyOnForms();
+
+        $fields[] = $password;
+
+        return $fields;
+    }
+
+    public function createNewFormBuilder(EntityDto $entityDto, KeyValueStore $formOptions, AdminContext $context): FormBuilderInterface
+    {
+        $formBuilder = parent::createNewFormBuilder($entityDto, $formOptions, $context);
+
+        return $this->addPasswordEventListener($formBuilder);
+    }
+
+    public function createEditFormBuilder(EntityDto $entityDto, KeyValueStore $formOptions, AdminContext $context): FormBuilderInterface
+    {
+        $formBuilder = parent::createEditFormBuilder($entityDto, $formOptions, $context);
+
+        return $this->addPasswordEventListener($formBuilder);
+    }
+
+    private function addPasswordEventListener(FormBuilderInterface $formBuilder): FormBuilderInterface
+    {
+        return $formBuilder->addEventListener(FormEvents::POST_SUBMIT, function ($event) {
+            $form = $event->getForm();
+            $data = $form->getData();
+            if (!$form->isValid()) {
+                return;
+            }
+
+            $password = $form->get('password')->getData();
+            if ($password) {
+                /* @phpstan-ignore-next-line */
+                $data->setPassword($this->userPasswordHasher->hashPassword($this->getUser(), $password));
```

Comment:
Quid plutôt de ça ?
```suggestion
                /** @var User */
                $user = $this->getUser();
                $data->setPassword($this->userPasswordHasher->hashPassword($user, $password));
```

En ajoutant `implements PasswordAuthenticatedUserInterface` à `User`

On pourrait dire que ça ajoute un import Symfony dans le domaine...

Mais toute cette admin ajoute un couplage + fort entre l'infra et le domaine de toute façon, et ça me semble mieux que ça apparaisse explicitement dans le code